docs/src/main/asciidoc/conditional-extension-dependencies.adoc

```diff
@@ -171,10 +173,61 @@ Conditional dependencies can also be configured in the Quarkus extension descrip
 
   <!-- SKIPPED CONTENT -->
 ----
+<1> the runtime Quarkus extension artifact ID;
+<2> the goal that generates the extension descriptor which every Quarkus runtime extension project should be configured with;
+<3> the conditional dependency configuration element;
+<4> the artifact coordinates of conditional dependencies on other extensions.
+
+In this case, the Maven dependency is not at all required in the `pom.xml` file.
+
+== Dev mode-only extension dependencies
+
+Extensions can also declare conditional dependencies on other extensions using Dev mode as the condition or one of the conditions for those dependencies to be activated.
+
+Dev mode-only extension dependencies can be configured in the Quarkus extension plugin in the following way:
+
+[source,xml]
+----
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+  <!-- SKIPPED CONTENT -->
+
+  <artifactId>quarkus-extension-a</artifactId> <1>
+
+  <!-- SKIPPED CONTENT -->
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-extension-maven-plugin</artifactId>
+        <version>${quarkus.version}</version>
+        <executions>
+          <execution>
+            <phase>process-resources</phase>
+            <goals>
+              <goal>extension-descriptor</goal> <2>
+            </goals>
+            <configuration>
+              <conditionalDevDependencies> <3>
+                <extension>org.acme:quarkus-extension-b:${b.version}</extension> <4>
+              </conditionalDevDependencies>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+
+  <!-- SKIPPED CONTENT -->
+----
+<1> the runtime Quarkus extension artifact ID;
+<2> the goal that generates the extension descriptor which every Quarkus runtime extension project should be configured with;
+<3> the Dev mode conditional dependency configuration element;
+<4> the artifact coordinates of conditional dependencies on extensions that should be evaluated only if an application is launched in Dev mode.
```

Comment:
I think the convention is to use `dev mode` without the capitalized `Dev`. Should probably be adjusted everywhere.